libs/agno/agno/workflow/v2/task.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Type, Union
+from uuid import uuid4
+
+from agno.agent import Agent
+from agno.run.response import RunResponse
+from agno.team import Team
+from agno.utils.log import logger
+
+
+@dataclass
+class Task:
+    """A single unit of work in a workflow sequence"""
+
+    name: str
+    executor: Union[Agent, Team]
+
+    task_id: Optional[str] = None
+    description: Optional[str] = None
+    expected_input: Optional[Dict[str, Type]] = None
+    expected_output: Optional[str] = None
+
+    # Task configuration
+    retry_count: int = 0
+    max_retries: int = 3
+    timeout_seconds: Optional[int] = None
+
+    skip_on_failure: bool = False
+
+    # Input validation mode
+    # If False, only warn about missing inputs
+    strict_input_validation: bool = False
+
+    def __post_init__(self):
+        if self.task_id is None:
+            self.task_id = str(uuid4())
+
+    def execute(self, inputs: Dict[str, Any], context: Dict[str, Any] = None) -> RunResponse:
```

Comment:
```suggestion
    def execute(self, input: TaskInput) -> RunResponse:
```
Should contain input query, context, workflow_session_state, images, etc.